### ISSUE TITLE
Does pretrain.py in code v2 support batch_size>1? Why do I get an error when I use Batch_size larger than 1

### ISSUE THREAD

**Issue by WUHU-G** [getao/icae#16]

**Comment by getao** [comment]
yes, v2 implementation now only supports batch=1 because we didn't observe throughput improvement after increasing batch size.

v1 doesn't have this constraint.

**Comment by RewindL** [comment]
> _No description provided._

Hello. Same issue occurs to me recently. Have you solved to train with batch_size > 1 on v2 implementation ?
I have read the source codes and i guess the problem is `mem_flag` and `decoder_mem_flag` in ICAE class serve only for batch_size=1 because it flatten the batch_dim when execute code like `segment_input_embedding[mem_flag]`, a better implementation might be using `torch.gather` function to maintain the dims or other ways to recover batch dim.
